dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/RequiredFunctionChoiceBehavior.cs

```diff
@@ -0,0 +1,163 @@
+    /// Fully qualified names of subset of the <see cref="Kernel"/>'s plugins' functions information to provide to the model.
+    /// </summary>
+    [JsonPropertyName("functions")]
+    public IEnumerable<string>? Functions
+    {
+        get => this._functionFQNs;
+        init
+        {
+            if (value?.Count() > 0 && this._functions?.Count() > 0)
+            {
+                throw new KernelException("Functions are already provided via the constructor.");
+            }
+
+            this._functionFQNs = value;
+        }
+    }
+
+    /// <summary>
+    /// The maximum number of function auto-invokes that can be made in a single user request.
+    /// </summary>
+    /// <remarks>
+    /// After this number of iterations as part of a single user request is reached, auto-invocation
+    /// will be disabled. This is a safeguard against possible runaway execution if the model routinely re-requests
+    /// the same function over and over. To disable auto invocation, this can be set to 0.
+    /// </remarks>
+    [JsonPropertyName("maximumAutoInvokeAttempts")]
+    public int MaximumAutoInvokeAttempts { get; init; } = DefaultMaximumAutoInvokeAttempts;
+
+    /// <summary>
+    /// Number of requests that are part of a single user interaction that should include this functions in the request.
+    /// </summary>
+    /// <remarks>
+    /// This should be greater than or equal to <see cref="MaximumAutoInvokeAttempts"/>.
+    /// Once this limit is reached, the functions will no longer be included in subsequent requests that are part of the user operation, e.g.
+    /// if this is 1, the first request will include the functions, but the subsequent response sending back the functions' result
+    /// will not include the functions for further use.
+    /// </remarks>
+    [JsonPropertyName("maximumUseAttempts")]
+    public int MaximumUseAttempts { get; init; } = 1;
+
+    /// <inheritdoc />
+    public override FunctionChoiceBehaviorConfiguration GetConfiguration(FunctionChoiceBehaviorContext context)
+    {
+        bool autoInvoke = this.MaximumAutoInvokeAttempts > 0;
+
+        // If auto-invocation is specified, we need a kernel to be able to invoke the functions.
+        // Lack of a kernel is fatal: we don't want to tell the model we can handle the functions
+        // and then fail to do so, so we fail before we get to that point. This is an error
+        // on the consumers behalf: if they specify auto-invocation with any functions, they must
+        // specify the kernel and the kernel must contain those functions.
+        if (autoInvoke && context.Kernel is null)
+        {
+            throw new KernelException("Auto-invocation for Required choice behavior is not supported when no kernel is provided.");
+        }
+
+        List<KernelFunction>? requiredFunctions = null;
+        bool allowAnyRequestedKernelFunction = false;
+
+        // Handle functions provided via constructor as function instances.
```

Comment:
Logic in this method is very similar to logic in `AutoFunctionChoiceBehavior.GetConfiguration` method. Should we move it to the base class and re-use?

Reply:
Fixed.